R/util.R

```diff
@@ -1711,3 +1780,82 @@ get_mode <- function(x, na.rm = FALSE) {
   ux <- unique(x)
   return(ux[which.max(tabulate(match(x, ux)))])
 }
+# Wrapper function that takes care of dplyr::group_by and dplyr::summarize as a single step.
+# @export
+summarize_group <- function(.data, grp_cols = NULL, grp_aggregations = NULL, ...){
+  library(dplyr)
+  if(length(grp_cols) == 0) {
+    .data %>% summarize(...)
+  } else {
+    groupby_args <- list() # default empty list
+    name_list <- list()
+    name_index = 1
+    # If group by columns and associated aggregation funcionts are provided,
+    # quote the columns/functions with rlang::quo so that dplyr can understand them.
+    if (!is.null(grp_cols) && !is.null(grp_aggregations)) {
+      groupby_args <- purrr::map2(grp_aggregations, grp_cols, function(func, cname) {
+        if(is.na(func) || length(func)==0 || func == "none"){
+          rlang::quo((UQ(rlang::sym(cname))))
+        } else if (func %in% c("fltoyear","rtoyear",
+            "fltohalfyear",
+            "rtohalfyear",
+            "fltoquarter",
+            "rtoq",
+            "fltobimonth",
+            "rtobimon",
+            "fltomonth",
+            "rtomon",
+            "fltoweek",
+            "rtoweek",
+            "fltoday",
+            "rtoday",
+            "year",
+            "halfyear",
+            "quarter",
+            "bimonth",
+            "bimon",
+            "month",
+            "mon",
+            "monthname",
+            "monname",
+            "monthnamelong",
+            "monnamelong",
+            "week",
+            "dayofyear",
+            "dayofquarter",
+            "dayofweek",
+            "day",
+            "wday",
+            "wdaylong",
+            "hour",
+            "minute",
+            "second")) {
+          # For date column, call extract_from_date
+          rlang::quo(extract_from_date(UQ(rlang::sym(cname)), type = UQ(func)))
+        } else if (func %in% c("asnum","asint","asintby10","aschar")) {
+          # For numeric column, call categorize_numeric
+          rlang::quo(categorize_numeric(UQ(rlang::sym(cname)), type = UQ(func)))
+        } else {
+          rlang::quo(UQ(func)(UQ(rlang::sym(cname))))
+        }
+      })
+      # create a name list
+      name_list <- purrr::map2(grp_aggregations, grp_cols, function(func, cname) {
+        if(is.na(func) || length(func)==0 || func == "none"){
+          cname
+        } else {
+          stringr::str_c(cname, func, sep = "_")
+        }
+      })
+      names(groupby_args) <- name_list
+      .data %>% dplyr::group_by(!!!groupby_args) %>% summarize(...)
+    } else {
+      if(!is.null(grp_cols)) {
```

Comment:
Short comment on what this case does would help.

Reply:
Added comment.